### ISSUE TITLE
🐞 - Combo resets search prematurely

### ISSUE THREAD

**Issue by evantrimboli** [Tinkoff/taiga-ui#2346]
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://taiga-ui.dev/components/combo-box#virtual

### Is this issue blocking you?

Non-Blocking

### Description

- Focus the countries box
- Type "Canad", wait.
- The list is filtered to only show Canada, which is correct.
- Now append "a" so the string reads "Canada".
- The list filtering has now been cleared and all options are visible.

Assume in my list I have the following values:
- Canada
- Canada 01
- Canada 02

By virtue of matching with the prefix, I never get the option to choose any other items that might be a partial match.

### Angular version

14.0.0

### Taiga UI version

2.57.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

**Comment by waterplea** [comment]
@splincode I think this issue needs a careful evaluation. Purely disabling strict matching will render people unable to select anything but `Canada`, unless they erase the input value, which is not obvious and arguably worse than just typing a space after `Canada` when you expect some other result or stopping at `Canad` and picking from the options. I don't want to introduce reliance on clicking/blurring the control to update some superfluous internal state of acceptance. Maybe there's a way we can incorporate `updateOn` option from Angular form controls?

**Comment by evantrimboli** [comment]
I'm not sure if there's necessarily a problem with it selecting the value at that point, however it would be nice if it didn't reset the options. That way the value is still selected, but you have the chance to continue typing/selecting for something more specific.

**Comment by waterplea** [comment]
But you can continue typing currently. It resets options because otherwise once you typed `Canada` you will not see anything but `Canada` unless you erase some of it. And that's not good UX. People should be aware of other options, people should be able to filter options by typing in ComboBox. Both of these things now work.

**Comment by evantrimboli** [comment]
You can continue to type, however:
- It just looks weird (opinion based, obviously)
- It triggers search change, which has impacts on remote filtering.

**Comment by waterplea** [comment]
I believe we specifically made it emit `null` in `searchChange` to signify that this is a match. Anyway, I'm keeping this open to think how we can improve this.

**Comment by sreekumarp** [comment]
May I take this item ?

as soon as we a match is found we should scroll to that particular item and show it as selected to the user,

![Image](https://github.com/user-attachments/assets/cc38263f-e627-480f-ad4f-c0bb965cdcac)

this would require the item to be found and render the new set of items

**Comment by waterplea** [comment]
@sreekumarp I don't think we should change the way comobobox works now, I'm keeping it open to keep in mind for the new version based on new textfield that we will be developing soon.
